3.4.14/docker-entrypoint.sh

```diff
@@ -3,7 +3,7 @@
 set -e
 
 # Allow the container to be started with `--user`
-if [[ "$1" = 'zkServer.sh' && "$(id -u)" = '0' ]]; then
+if [[ "$1" = 'zkServer.sh' && "$(id -u)" = '0' && "$ZOO_USER" != 'root' ]]; then
```

Comment:
Could you please explain the logic behind this change? The original idea was to change a user to non-priveleged only if the user is not overridden by the `--user` flag (`"$(id -u)" = '0'` meaning we're running as `root`). But you've added a check that `$ZOO_USER` is not `root`, which is a bit confusing.

Reply:
Could you please explain why do you need to run Zookeeper within a container as `root`? This isn't a good security practice. Hence it's unlikley that this official image would support that.

Also, `ZOO_USER` wasn't meant to be overriden, that's why it's not documented.